exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,305 @@
+    if (!connect_active_) {
+      connect_active_ = true;
+      co_await asio::co_spawn(mqtt_client_->strand(), connect_to_broker(), asio::use_awaitable);
+    }
+
+    // When a new list of banned string arrives the program is restarted
+    config_.value()._banned_topics.observe([this](auto& new_conf, [[maybe_unused]] auto& old_conf) {
+      banned_signals_.clear();
+      for (auto const& con : new_conf) {
+        banned_signals_.push_back(con);
+      }
+      restart();
+    });
+
+    asio::co_spawn(mqtt_client_->strand(), tfc::base::exit_signals(ctx_), asio::detached);
+
+    banned_signals_ = config_.value()._banned_topics.value();
+    load_signals();
+  }
+
+  // Connect the mqtt client to the broker. Only a single "thread" can run this at a time. The routine that calls this
+  // function must set the connect_active_ flag to true before calling this function and the function sets it to false when
+  // finished.
+  auto connect_to_broker() -> asio::awaitable<void> {
+    while (true) {
+      try {
+        logger_.trace("Connecting to MQTT broker");
+        asio::ip::tcp::socket resolve_sock{ ctx_ };
+        asio::ip::tcp::resolver res{ resolve_sock.get_executor() };
+
+        logger_.trace("Resolving IP number");
+        asio::ip::tcp::resolver::results_type resolved_ip =
+            co_await res.async_resolve(mqtt_host_, mqtt_port_, asio::use_awaitable);
+
+        logger_.trace("Connecting MQTT client socket to IP number of broker");
+        co_await asio::async_connect(mqtt_client_->next_layer(), resolved_ip, asio::use_awaitable);
+
+        logger_.trace("Sending Connect request to the MQTT broker");
+
+        // Connect is initialized with clean start set to false, which means that the broker will keep the session alive even
+        // if the client disconnect, and it will keep that session alive for UINT_MAX seconds.
+        auto connect_packet =
+            async_mqtt::v5::connect_packet{ false,
+                                            std::chrono::seconds(100).count(),
+                                            async_mqtt::allocate_buffer("cid1"),
+                                            async_mqtt::nullopt,
+                                            async_mqtt::allocate_buffer(mqtt_username_),
+                                            async_mqtt::allocate_buffer(mqtt_password_),
+                                            { async_mqtt::property::session_expiry_interval{ UINT_MAX } } };
+
+        co_await mqtt_client_->send(connect_packet, asio::use_awaitable);
+
+        logger_.trace("Waiting for MQTT connection acknowledgement");
+        co_await mqtt_client_->recv(async_mqtt::filter::match, { async_mqtt::control_packet_type::connack },
+                                    asio::use_awaitable);
+
+        connect_active_ = false;
+        co_return;
+      } catch (std::exception& e) {
+        logger_.error("Error while connecting to MQTT broker: {}", e.what());
+      } catch (...) {
```

Comment:
does this ever occur? because you are catching C any arg exception? 
I think the std::exception is enough

Reply:
find it unlikely, changing to only std::exception.